addons/website_sale/controllers/main.py

```diff
@@ -730,23 +725,46 @@ def address(self, **kw):
                 if not errors:
                     return request.redirect(kw.get('callback') or '/shop/confirm_order')
 
-        country = 'country_id' in values and values['country_id'] != '' and request.env['res.country'].browse(int(values['country_id']))
-        country = country and country.exists() or def_country_id
         render_values = {
             'website_sale_order': order,
             'partner_id': partner_id,
             'mode': mode,
             'checkout': values,
             'can_edit_vat': can_edit_vat,
-            'country': country,
-            'country_states': country.get_website_sale_states(mode=mode[1]),
-            'countries': country.get_website_sale_countries(mode=mode[1]),
             'error': errors,
             'callback': kw.get('callback'),
-            'only_services': order and order.only_services,
+            'only_services': order and order.only_services
         }
+        render_values.update(self._get_country_related_render_values(kw, render_values))
         return request.render("website_sale.address", render_values)
 
+    def _get_country_related_render_values(self, kw, render_values):
```

Comment:
@JKE-be Is this refactor (stable policy compatible) ok for you?

Reply:
not the most elegant but yes it is ok for stable policy